src/app/projects/components/pagination/pagination.component.ts

```diff
@@ -35,12 +36,34 @@ export class PaginationComponent implements OnInit, OnChanges {
 
   ngOnChanges(changes: SimpleChanges): void {
     this.calculateTotalPages();
+    this.calculateVisibleButtons();
   }
 
   private calculateTotalPages() {
     this.totalPages = Math.ceil(this.totalItems / this.itemsPerPage);
   }
 
+  private calculateVisibleButtons() {
+    switch (this.currentPage) {
+      case 1:
+        this.visibleButtons = [1, 2, 3];
+        break;
+      case this.totalPages:
```

Comment:
What happens if the `totalPages` is 2?
Then we would have: `this.visibleButtons = [ 0, 1, 2]`
Is that ok?

Reply:
No it won't. Look in the template, it is hiding if below 1 or above totalpages

Reply:
I guess this is a little unobvious but it will be a lot of cases in this switch case if I don't do it this way... I think